api/controllers/user.js

```diff
@@ -164,6 +164,27 @@ const removeProjectPermission = async (user, projId) => {
   });
 }
 
+const removeUser = async (user) => {
+  const UserModel = mongoose.model('User');
+  const ProjectModel = mongoose.model('Project');
+  if (user._id && user.displayName) {
+    defaultLog.info(`Attempting to remove user ${user.displayName}`);
+    try {
+      const result = await UserModel.findOneAndRemove({ _id: user._id });
```

Comment:
```suggestion
      const result = await UserModel.findOneAndDelete({ _id: user._id}, { useFindAndModify: false });
```
- `findOneAndDelete()` should be used in place of `findOneAndRemove()`, since the former uses the new mongodb `findOneAndDelete` command, whereas the latter replicates the functionality through the slower `findAndModify` remove command, which is deprecated. See [here](https://mongoosejs.com/docs/5.x/docs/deprecations.html#remove) for more details.
- The `useFindAndModify` setting should be set to `false` when using `findOneAndDelete` in Mongo 5.x for the same reason, unless you set the option globally:
`mongoose.set('useFindAndModify', false);`

Reply:
Sounds good, thanks for pointing that out. I'll update this instance for now, but there are also 7 other instances in the API that we should update in the future.